### ISSUE TITLE
CSFR Token issue when Phoenix.Html.Link.button/2 is rendered in a partial

### ISSUE THREAD

**Issue by guidotripaldi** [grych/drab#130]
I've noticed that when a (non Drab) button is rendered by `Phoenix.Html.Link.button/2` in a partial, the CSFR token isn't anymore valid after poking some assignments in the main template. Maybe it is related to #67 because according to the Phoenix documentation the button is rendered by Phoenix using a regular HTML form. 
This is the test code:

```eex
# main.html.drab
<button type="button" drab="click:update_assignments">Do Poke Assignment</button>
<%= inspect @foo %>

<%= render("_sub.html", assigns) %>
```

```eex
# _sub.html.eex
   <%= button("Do My Action", to: my_path(@conn, :my_action) ) %>
```

```elixir
# Commander

defhandler poke_assignment(socket, sender) do
   poke socket, foo: 42
end
```

If the "Do My Action" button is clicked before pressing the "Do Poke Assignment" button it works well, but after poking a value and press the button again, it raise the exception:

```invalid CSRF (Cross Site Request Forgery) token, make sure all requests include a valid '_csrf_token' param or 'x-csrf-token' header```

Workaround seems to be difficult because its not clear for me how to access the CSFR token, as suggested in #67 to save and set back the token, because the 'form' whose belong is not explicit.

EDIT:
Fixed some errors in the problem description

**Comment by grych** [comment]
It is related to #67, actually it is very similar.

Phoenix `button/2` generates something like:

```html
<button data-csrf="PjcXYRgmBAEKFDIHNz8wWy8TEx0vAAAAjxnVPjkYaBdURIY+JwFTUw==" data-method="post" data-to="/tests/live">...</button>
```

So when you rerender it, token changes and Phoenix complains about it.

As a workaround, you may get the token before:

```elixir
 %{"dataset" => %{"csrf" => old_csrf}} = query_one! socket, "#a", "dataset"
```

and set it after poke:
```elixie
set_data socket, "#a", csrf: old_csrf
```
The ideal would be if Drab could store those tokens itself... Lemme think about it.

**Comment by guidotripaldi** [comment]
What do you prefer, that I left this case open, as Drab has to be enhanced to circumvent this limitation, or I close it because there is a working workaround available?

**Comment by grych** [comment]
Please leave it open.

I think it should be solved somehow before 1.0. This is because I want Drab to be as compatible with Phoenix as it is possible.

Also, Drab is targeted to the beginners as well. The idea is to make everything easier, and such workarounds are not helping at all :)

This is a rough plan for a solution:
- drab.live.js detects all csrf tokens (in buttons and in hidden inputs in forms) and stores them internally (issue: what about buttons without an `id`?)
- drab.live.js sends this information to server
- after re-render the html, Drab.Live updates all elements with tokens with the saved ones; ideally it would be if it detects where to update (and not in the whole document)
